projects/packages/videopress/src/class-access-control.php

```diff
@@ -0,0 +1,355 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+/**
+ * VideoPress Access Control.
+ *
+ * @package automattic/jetpack-videopress
+ */
+
+namespace Automattic\Jetpack\VideoPress;
+
+use Automattic\Jetpack\Connection\Client;
+use VIDEOPRESS_PRIVACY;
+
+/**
+ * VideoPress video access control utilities.
+ *
+ * Note: this is also being used on WordPress.com.
+ * Use IS_WPCOM checks for functionality that is specific to WPCOM/Jetpack.
+ */
+class Access_Control {
+
+	/**
+	 * Singleton Access_Control instance.
+	 *
+	 * @var Access_Control
+	 **/
+	private static $instance = null;
+
+	/**
+	 * Guid to subscription plan, store, for when used inline on a page.
+	 *
+	 * @var array
+	 */
+	private $guids_to_subscriptions = array();
+
+	/**
+	 * Set that this guid is controlled by a subscription.
+	 *
+	 * @param string     $guid            The guid to set.
+	 * @param string|int $subscription_id The subscription to set.
+	 *
+	 * @return Access_Control
+	 */
+	public function set_guid_subscription( $guid, $subscription_id ) {
+		$this->guids_to_subscriptions[ $guid ] = $subscription_id;
+		return $this;
+	}
+
+	/**
+	 * Get the subscription for a guid.
+	 *
+	 * @param string $guid The guid to get.
+	 *
+	 * @return string|int|false
+	 */
+	public function get_subscription_plan_id( $guid ) {
+		return isset( $this->guids_to_subscriptions[ $guid ] ) ? $this->guids_to_subscriptions[ $guid ] : false;
+	}
+
+	/**
+	 * Get the singleton instance.
+	 *
+	 * @return AJAX
+	 */
+	public static function instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Determines if Jetpack Memberships are available.
+	 *
+	 * @return bool
+	 */
+	private function jetpack_memberships_available() {
+		return class_exists( '\Jetpack_Memberships' );
+	}
+
+	/**
+	 * Determines if Jetpack Subscriptions are available.
+	 *
+	 * @return bool
+	 */
+	private function jetpack_subscriptions_available() {
```

Comment:
This function relies on checking for existing functions. I wonder if it should rely on existing utilities managing editor extension availability, as they're defined in `Jetpack_Gutenberg`, such as `is_available()`?

Should it also take module availability into account, and check if the Subscriptions module is active for example?

Reply:
My reasons for checks I think I covered in the other https://github.com/Automattic/jetpack/pull/32154/files/16f380bd348c72695508a82d8501c9ee82a49959#r1354597036. 

> Should it also take module availability into account, and check if the Subscriptions module is active for example?

Maybe, let me check and report back